include/PriorityQueue.h

```diff
@@ -9,8 +9,32 @@ class PriorityQueue {
 
 public:
     void push(const T &t);
+
     T poll();
 
 private:
+    struct {
+        T operator()(T a, T b) const {
+            return a - b;
+        }
+    } m_comp;
+
+    std::list<T> m_pq;
 };
 
+template<class T>
+void PriorityQueue<T>::push(const T &t) {
```

Comment:
There no need to separate the method declaration and the method definition. You could just put them together.